base/acme/src/main/java/org/dogtagpki/acme/issuer/PKIIssuer.java

```diff
@@ -77,7 +80,40 @@ public void init() throws Exception {
             clientConfig.setUsername(username);
         }
 
-        String password = config.getParameter("password");
+        String password = null;
+        String passwordFilename = config.getParameter("passwordFile");
+        if (passwordFilename != null) {
+            // read password from file
+            File passwordFile = new File(passwordFilename);
+
+            if (passwordFile.canRead()) {
+                logger.info("Loading password file " + passwordFilename);
+                Properties props = new Properties();
+                try (FileReader reader = new FileReader(passwordFile)) {
+                    props.load(reader);
+                }
+                password = props.getProperty("acmeUserPassword");
+                if (password == null) {
+                    throw new RuntimeException(
+                        "PKIIssuer: 'acmeUserPassword' property not present in file "
+                        + passwordFilename
+                    );
+                }
+            } else {
+                throw new RuntimeException(
+                    "PKIIssuer: password file '" + passwordFilename + "' "
+                    + "cannot be read or does not exist."
+                );
+            }
+        } else {
+            password = config.getParameter("password");
+            if (password == null) {
```

Comment:
The `PKIIssuer` acctually supports client cert authentication, so the user password is not required.

Reply:
Oh, I didn't realise, but I see it now.

But, how then do you specify the nssdb, nssdb password and token name?

Reply:
In normal CA installation, Tomcat will have a `PKIListener` (which is a subclass of `JSSListener`) in server.xml which will initialize the NSS database on startup. The NSS database and password file can be specified either in jss.conf or in a `<Connector>` in server.xml. See:
https://github.com/dogtagpki/tomcatjss/blob/master/src/org/dogtagpki/tomcat/JSSListener.java#L51

If ACME is installed in the same instance, it will use the same NSS database and have access to NSS tokens since it's already initialized by `JSSListener`. ACME currently doesn't support token name for client cert, but we can add another parameter for that. So for now the client cert has to be added into the internal token of the server's NSS database.

Alternatively, ACME can be installed on a standalone Tomcat that doesn't have a CA:
https://www.dogtagpki.org/wiki/Installing_Standalone_ACME_Responder
In this case Tomcat will have a `JSSListener` instead of `PKIListener`, but it works the same way.